ui/components/ui/token-list-display/token-list-display.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { isEqual } from 'lodash';
+
+import { getShouldHideZeroBalanceTokens } from '../../../selectors';
+import { useTokenTracker } from '../../../hooks/useTokenTracker';
+import Identicon from '../identicon';
+import TokenBalance from '../token-balance';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getTokens } from '../../../ducks/metamask/metamask';
+
+export default function TokenListDisplay(props) {
+  const t = useI18nContext();
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+
+  const { clickHandler } = props;
+
+  const handleSelectToken = (token) => clickHandler(token);
+
+  const tokens = useSelector(getTokens, isEqual);
+  const { loading, tokensWithBalances } = useTokenTracker(
+    tokens,
+    true,
+    shouldHideZeroBalanceTokens,
+  );
+  if (loading) {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          height: '250px',
+          alignItems: 'center',
+          justifyContent: 'center',
+          padding: '30px',
+        }}
+      >
+        {t('loadingTokens')}
+      </div>
+    );
+  }
+
+  const sendableTokens = tokensWithBalances.filter((token) => !token.isERC721);
+
+  return (
+    <>
+      {sendableTokens.map((tokenData) => {
+        const { address, image, symbol } = tokenData;
+
+        return (
+          <div
+            key={address}
+            className="send-v2__asset-dropdown__asset"
+            onClick={() => handleSelectToken(tokenData)}
+          >
+            <div className="send-v2__asset-dropdown__asset-icon">
```

Comment:
[REQ] If we are going to extract these into a UI component we need to update the class names to not reference the send flow, and we should use the <Box> component where it makes sense as well as <Typography> to reduce stylesheet clutter.